### PR TITLE
Move DAML snippets from inline to their own files, so they're tested

### PR DESCRIPTION
**Pull request by bethaitman-da**
These were all the ones I could move trivially. The ones I haven't done this time are:

- [daml/1-2-conversion.rst](https://github.com/digital-asset/daml/blob/master/docs/source/daml/1-2-conversion.rst), because there are ton of them, and half of them are 1.0 snippets
- [daml/reference/updates.rst](https://github.com/digital-asset/daml/blob/master/docs/source/daml/reference/updates.rst), because they are pseudocode and it would take a bit of rewriting so they're real
- [daml/reference/expressions.rst](https://github.com/digital-asset/daml/blob/master/docs/source/daml/reference/expressions.rst), because there's only one and it would take writing some logic around it
- [daml/sdlib/index.rst](https://github.com/digital-asset/daml/blob/master/docs/source/daml/stdlib/index.rst) because I've got a separate PR to tidy up this page

### CODE REVIEW

File: docs/source/app-dev/bindings-java/code-snippets/ProductTypes.daml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- start snippet: product types example
+daml 1.2
+module Com.Acme where
+
+data Person = Person with name : Name; age : Decimal
+data Name = Name with firstName : Text; lastName : Text
+-- end snippet: product types example
```
Comment (neil-da) [inline docs/source/app-dev/bindings-java/code-snippets/ProductTypes.daml]: I recommend configuring your editor to add trailing newlines.

File: docs/source/app-dev/bindings-java/code-snippets/Templates.daml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+daml 1.2
+module Com.Acme where
+
+template Bar
+  with
+    owner: Party
+    name: Text
+  where
+    controller owner can
+      Bar_SomeChoice: (Bool)
```
Comment (neil-da) [inline docs/source/app-dev/bindings-java/code-snippets/Templates.daml]: Generally brackets around `(Bool)` are not necessary

Comment (neil-da) [inline docs/source/app-dev/bindings-java/code-snippets/Templates.daml]:
```suggestion
      Bar_SomeChoice: Bool
```

File: docs/source/daml/daml-studio.rst [hunk 3]
```diff
@@ -223,6 +223,10 @@ Missing authorization on create
 If a contract is being created without approval from all authorizing
 parties the commit will fail. For example:
 
+.. literalinclude:: daml-studio/daml/CreateAuthFailure.daml
+  :language: daml
+  :lines: 7-13,15-18
+
```
Comment (neil-da) [inline docs/source/daml/daml-studio.rst]: This seems like duplicating the same lump twice?

File: docs/source/app-dev/bindings-java/code-snippets/ProductTypes.daml (hunk 1 above)
Comment (bethaitman-da) [reply docs/source/app-dev/bindings-java/code-snippets/ProductTypes.daml]: fixed!

File: docs/source/daml/daml-studio.rst (hunk 3 above)
Comment (bethaitman-da) [reply docs/source/daml/daml-studio.rst]: testing something and mistakenly left this in - gone now

**Pull request merged**